src/main/java/io/cdap/plugin/switchcase/route/RoutingSwitch.java

```diff
@@ -0,0 +1,348 @@
+        // defaulting value, based on the selected default handling option
+        emitDefaultValue(input, emitter);
+      } else {
+        emitNullValue(input, emitter);
+      }
+    }
+    // if a port is found, emit to the port
+    emitter.emit(port, input);
+  }
+
+  private void emitDefaultValue(StructuredRecord input, MultiOutputEmitter<StructuredRecord> emitter) {
+    if (Config.DefaultHandling.DEFAULT_PORT == config.getDefaultHandling()) {
+      emitter.emit(config.defaultPort, input);
+    } else if (Config.DefaultHandling.ERROR_PORT == config.getDefaultHandling()) {
+      String error = String.format("Record contained unknown value for field %s", config.routingField);
+      InvalidEntry<StructuredRecord> invalid = new InvalidEntry<>(1, error, input);
+      emitter.emitError(invalid);
+    } else if (Config.DefaultHandling.SKIP == config.getDefaultHandling()) {
+      LOG.trace("Skipping record because value for field {} did not match any rule in the port specification",
+                config.routingField);
+    }
+  }
+
+  private void emitNullValue(StructuredRecord input, MultiOutputEmitter<StructuredRecord> emitter) {
+    if (Config.NullHandling.NULL_PORT == config.getNullHandling()) {
+      emitter.emit(config.nullPort, input);
+    } else if (Config.NullHandling.ERROR_PORT == config.getNullHandling()) {
+      String error = String.format("Record contained null value for field %s", config.routingField);
+      InvalidEntry<StructuredRecord> invalid = new InvalidEntry<>(2, error, input);
+      emitter.emitError(invalid);
+    } else if (Config.NullHandling.SKIP == config.getNullHandling()) {
+      LOG.trace("Skipping record because field {} is null", config.routingField);
+    }
+  }
+
+  private Map<String, Schema> generateSchemas(Schema inputSchema) {
+    Map<String, Schema> schemas = new HashMap<>();
+    // Add all ports from the port config to the schemas
+    for (String port : evaluator.getAllPorts()) {
+      schemas.put(port, inputSchema);
+    }
+    // If defaulting records need to be sent to their own port, add that port to the schemas
+    if (Config.DefaultHandling.DEFAULT_PORT == config.getDefaultHandling()) {
+      schemas.put(config.defaultPort, inputSchema);
+    }
+    // If null values need to be sent to their own port, add that port to the schemas
+    if (Config.NullHandling.NULL_PORT == config.getNullHandling()) {
+      schemas.put(config.nullPort, inputSchema);
+    }
+    return schemas;
+  }
+
+  /**
+   * {@link PluginConfig} for {@link RoutingSwitch}
+   */
+  static class Config extends PluginConfig {
+    private static final List<Schema.Type> ALLOWED_TYPES = new ArrayList<>();
+    static {
+      ALLOWED_TYPES.add(Schema.Type.STRING);
+      ALLOWED_TYPES.add(Schema.Type.INT);
+      ALLOWED_TYPES.add(Schema.Type.LONG);
+      ALLOWED_TYPES.add(Schema.Type.FLOAT);
+      ALLOWED_TYPES.add(Schema.Type.DOUBLE);
+      ALLOWED_TYPES.add(Schema.Type.BOOLEAN);
```

Comment:
Decimal ? Time / Timestamp

Reply:
yeah, actually we need to add more functions. even the numeric and boolean types are kind of useless right now. I'll create a separate PR to add more functions for these data types. Currently all the functions apply on strings.

Reply:
Could you please file a jira for it and add it as TODO. 
Also if we are not going to add it immediately its worth adding it in the plugin documentation.